### PR TITLE
[QAP-492] Automate API of Admin Panel - Shifts page

### PR DESCRIPTION
**Pull request by OlehBright**

### CODE REVIEW

File: pages/Api.ts [hunk 1]
```diff
@@ -66,6 +68,7 @@ export class Api {
         this.internalPage = new ApiInternalPage(apiContext)
         this.otherTestsPage = new ApiOtherPage(apiContext)
         this.agora3002Page = new Api3002Page(apiContext)
+        this.adminPanelPage = new AdminPanelPage(apiContext)
```
Comment (IvanK-QX) [inline pages/Api.ts:71]: @OlehBright change spelling from AdminPanelPage to Api3011ShiftsPage

File: pages/apiPages/admin_panel_page.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        console.log(`Moderators list is received`)
+    }
+
+    async getModeratorsOnShift(url: string, userToken: string, moderatorId: string) {
+        const apiContext = await request.newContext({ignoreHTTPSErrors: true})
+        const data = {}
+        const headers = Headers.userHeader(userToken)
+        const apiRequest = await apiContext.get(`${url}:3011/admin/moderators/shift/ids`, {data, headers: headers})
+        expect(apiRequest.ok()).toBeTruthy()
+        const responseJson = await apiRequest.json()
```
Comment (IvanK-QX) [inline pages/apiPages/admin_panel_page.ts:34]: @OlehBright all responseJson to response

File: pages/apiPages/admin_panel_page.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        const role = responseJson.documents[3].rolesGroup
+        const name = responseJson.documents[3].about
+        expect(role).toEqual('moderator')
+        expect(name).toEqual('string')
+        console.log(`Moderators list is received`)
+    }
+
+    async getModeratorsOnShift(url: string, userToken: string, moderatorId: string) {
+        const apiContext = await request.newContext({ignoreHTTPSErrors: true})
+        const data = {}
```
Comment (IvanK-QX) [inline pages/apiPages/admin_panel_page.ts:30]: @OlehBright remove cons data

File: pages/apiPages/admin_panel_page.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        const headers = Headers.userHeader(userToken)
+        const apiRequest = await apiContext.get(`${url}:3011/admin/moderators/shift/ids`, {data, headers: headers})
```
Comment (IvanK-QX) [inline pages/apiPages/admin_panel_page.ts:32]: @OlehBright remove data from request parameters

File: pages/apiPages/admin_panel_page.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        expect(apiRequest.ok()).toBeTruthy()
+        const responseJson = await apiRequest.json()
+        expect(responseJson).toContain(moderatorId)
+        console.log(`Moderators on Shift list is received`)
+    }
+
+    async getModeratorsOnOtherShift(url: string, userToken: string, moderatorId: string) {
+        const apiContext = await request.newContext({ignoreHTTPSErrors: true})
+        const data = {}
```
Comment (IvanK-QX) [inline pages/apiPages/admin_panel_page.ts:41]: @OlehBright please remove `const data = {}` from all Get requests
also remove data variables from all get apiRequest

File: pages/apiPages/admin_panel_page.ts [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+        const headers = Headers.userHeader(userToken)
+        const apiRequest = await apiContext.get(`${url}:3011/admin/moderators/mp/shift/ids`, {data, headers: headers})
+        expect(apiRequest.ok()).toBeTruthy()
+        const responseJson = await apiRequest.json()
+        expect(responseJson).toContain(moderatorId)
```
Comment (IvanK-QX) [inline pages/apiPages/admin_panel_page.ts:46]: @OlehBright please add all request to Postman, so we can validate response properly

File: pages/apiPages/admin_panel_page.ts [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+        }
+        const headers = Headers.userHeader(userToken)
+        const apiRequest = await apiContext.post(`${url}:3011/admin/moderators/shift/start`, {data, headers: headers})
+        expect(apiRequest.ok()).toBeTruthy()
+        const responseJson = await apiRequest.json()
+        expect(responseJson).toContain(moderatorId)
+        console.log(`Moderator Other Shift has started`)
+    }
+
+
```
Comment (IvanK-QX) [inline pages/apiPages/admin_panel_page.ts:102]: @OlehBright remove useless strings

File: pages/apiPages/admin_panel_page.ts [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+        const responseJson = await apiRequest.json()
+        expect(responseJson).not.toContain(moderatorId)
+        console.log(`Moderator Safe Shift has started`)
+    }
+
+    async startOtherShiftForModerator(url: string, userToken: string, moderatorId: string) {
+        const apiContext = await request.newContext({ignoreHTTPSErrors: true})
+        const data = {
+            "moderatorId": `${moderatorId}`,
+            "isMpStreams": true
```
Comment (IvanK-QX) [inline pages/apiPages/admin_panel_page.ts:92]: @OlehBright this parameter should be in response and can be validated

File: pages/apiPages/admin_panel_page.ts [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+        const responseJson = await apiRequest.json()
+        expect(responseJson).not.toContain(moderatorId)
+        console.log(`Moderators on "Safe" Shift isn't present in "Other" shift list`)
+    }
+
+    async startSafeShiftForModerator(url: string, userToken: string, moderatorId: string) {
+        const apiContext = await request.newContext({ignoreHTTPSErrors: true})
+        const data = {
+            "moderatorId": `${moderatorId}`,
+            "isMpStreams": false
```
Comment (IvanK-QX) [inline pages/apiPages/admin_panel_page.ts:65]: @OlehBright this parameter should be in response and can be validated

File: pages/apiPages/admin_panel_page.ts [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+        const apiContext = await request.newContext({ignoreHTTPSErrors: true})
+        const data = {
+            "moderatorId": `${moderatorId}`
+        }
+        const headers = Headers.userHeader(userToken)
+        const apiRequest = await apiContext.post(`${url}:3011/admin/moderators/shift/end`, {data, headers: headers})
+        expect(apiRequest.ok()).toBeTruthy()
+        const responseJson = await apiRequest.json()
+        expect(responseJson).not.toContain(moderatorId)
+        console.log(`Moderator Safe Shift has started`)
```
Comment (IvanK-QX) [inline pages/apiPages/admin_panel_page.ts:85]: @OlehBright maybe finished?

**Review by IvanK-QX — CHANGES_REQUESTED** [review]

File: pages/apiPages/admin_panel_page.ts (hunk 8 above)
Comment (OlehBright) [reply pages/apiPages/admin_panel_page.ts]: isMpStreams parameter isn't returned in response. Can't be validated

File: pages/apiPages/admin_panel_page.ts (hunk 9 above)
Comment (OlehBright) [reply pages/apiPages/admin_panel_page.ts]: isMpStreams parameter isn't returned in response. Can't be validated

File: tests/api_tests/3010/admin_panel_test.spec.ts [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+
+let admin, newAdmin, createdNewAdmin
+
+test.describe('Admin Panel API test',async () => {
+    test.beforeEach(async () => {
+        const apiContext = await request.newContext()
+        const api = new Api(apiContext)
+        admin = await api.loginPage.loginWithAdminUser(apiUrl.qaEnvUrl)
+        const guestUser = await api.loginPage.login(`${apiUrl.qaEnvUrl}:3000/login`)
+        createdNewAdmin = await api.moderatorPage.createNewModerator(apiUrl.qaEnvUrl, admin.adminToken, apiDataSet.randomEmail)
```
Comment (IvanK-QX) [inline tests/api_tests/3010/admin_panel_test.spec.ts:14]: @OlehBright we should use admin here

File: tests/api_tests/3010/admin_panel_test.spec.ts [hunk 12]
```diff
@@ -0,0 +1,36 @@
+import { request, test } from "@playwright/test";
+import { apiUrl } from "../../../utils/apiUrl";
+import { Api } from "../../../pages/Api";
+import { apiDataSet } from "../../../utils/dataSet";
+
+let admin, newAdmin, createdNewAdmin
```
Comment (IvanK-QX) [inline tests/api_tests/3010/admin_panel_test.spec.ts:6]: @OlehBright remove createdNewAdmin , newAdmin

File: tests/api_tests/3010/admin_panel_test.spec.ts [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+
+test.describe('Admin Panel API test',async () => {
+    test.beforeEach(async () => {
+        const apiContext = await request.newContext()
+        const api = new Api(apiContext)
+        admin = await api.loginPage.loginWithAdminUser(apiUrl.qaEnvUrl)
```
Comment (IvanK-QX) [inline tests/api_tests/3010/admin_panel_test.spec.ts:12]: @OlehBright use const instead let 
const adminLogin =

**Review by IvanK-QX — CHANGES_REQUESTED** [review]

**Review by IvanK-QX — APPROVED** [review]

**Pull request merged**